garden-backend-service/src/modal/utils.py

```diff
@@ -76,6 +81,28 @@ async def monitor_modal_invocation(
                     raise e
                 # Otherwise, write the error to the DB
                 inv.error = str(e)
-                status = InvocationStatus.ERROR
+                status = AsyncModalJobStatus.ERROR
                 await resolve_modal_invocation(inv, status, session)
                 await session.commit()
+
+
+async def monitor_modal_deployment(
+    deploy_func: Callable,
+    deploy_config: dict,
+    db_modal_app: ModalApp,
+    settings: Settings,
+):
+    session_maker = await get_db_session_maker(settings=settings)
+
+    try:
+        deploy_func(deploy_config)
+        async with session_maker() as session:
+            if modal_app := await ModalApp.get(session, id=db_modal_app.id):
+                modal_app.deploy_status = AsyncModalJobStatus.DONE
+                await session.commit()
+    except Exception as e:
```

Comment:
understanding check: do we not want to re-raise this exception because it's problematic to raise an exception from a background task?

Reply:
if so I think it's a good idea to log some information about the exception (since it won't get picked up by the exception logging middleware)